stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/images/AttachmentGallery.kt

```diff
@@ -9,14 +11,33 @@ import androidx.core.content.res.use
 import androidx.core.view.isVisible
 import androidx.fragment.app.FragmentActivity
 import androidx.viewpager2.widget.ViewPager2
+import com.getstream.sdk.chat.ImageLoader
+import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
 import io.getstream.chat.android.ui.R
 import io.getstream.chat.android.ui.databinding.StreamUiAttachmentGalleryBinding
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 
 public class AttachmentGallery : ConstraintLayout {
 
+    private var onSharePictureListener: (pictureUri: Uri) -> Unit = { pictureUri ->
+        ContextCompat.startActivity(
+            context,
+            Intent.createChooser(
+                Intent(Intent.ACTION_SEND).apply {
+                    type = "image/*"
+                    putExtra(Intent.EXTRA_STREAM, pictureUri)
+                    addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+                    data = pictureUri
+                },
```

Comment:
I think there's _something_ wrong with this intent, but I couldn't debug the cause of the issues.

This is what I'm seeing with different target apps:
- Telegram: it works okay, but it looks like I've shared a file and not an image
- Telegram X: broken, toast appears with "this type of media is not yet supported"
- Messenger Lite: does nothing at all (doesn't even go the app, just closes the share sheet)
- Flamingo (Twitter client): opens a new tweet but the image is missing from it

Reply:
`setType` and `setData` are mutually exclusive, so `setData` overrides the mime type: 

```
    public @NonNull Intent setType(@Nullable String type) {
        mData = null;
        mType = type;
        return this;
    }
    
        public @NonNull Intent setData(@Nullable Uri data) {
        mData = data;
        mType = null;
        return this;
    }
```
Try removing `data = pictureUri`

Reply:
At least without `data = pictureUri` I see that image is shared as an image with Telegram

Reply:
Nice catch, removing that fixes all of my sharing issues with the apps I'm testing with indeed. Also still works on both my Android 10 and Android 11 devices.